src/main/java/com/shopping_app/shoppingApp/model/Response/AbstractResponse.java

```diff
@@ -6,13 +6,15 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import lombok.experimental.SuperBuilder;
 import org.springframework.http.HttpStatus;
 
 @Getter
 @Setter
 @NoArgsConstructor
 @AllArgsConstructor
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@SuperBuilder(builderMethodName = "baseBuilder")
 public class AbstractResponse {
```

Comment:
what purpose does this class serve?

Reply:
this class will act like abstract for all sub class for response , all the duplicate fields (repeating fields) 
are added in this  class